### PR TITLE
Iiscrum 89 zabbix

### PR DESCRIPTION
**Pull request by PeterJokinen**
Zabbix changes for testing tool, to easily recreate it's patch file. See PR [here ](https://github.com/digiaiiris/iiris/pull/1018)for more.

### CODE REVIEW

File: frontends/php/items.php [hunk 1]
```diff
@@ -866,6 +869,74 @@
...
+		]);
+		$item = $items[0];
+		$item['test_value'] = getRequest('test_value');
+		$item['test_delay'] = getRequest('test_delay');
+
+		DB::checkValueTypes('item_testing',$item);
+		$error = false;
+		try {
+			$sql = 'INSERT INTO item_testing ('.implode(',', array_keys($item)).')'.
+					' VALUES ('.implode(',', array_values($item)).')';
```
Comment (sapajune) [inline frontends/php/items.php]: Does this work for text values? I would assume text values should be in single quotes.

Also, this should probably include some SQL injection prevention.

File: frontends/php/items.php [hunk 2]
```diff
@@ -866,6 +869,74 @@
...
+			unset($_REQUEST['itemid'], $_REQUEST['form']);
+		}
+	}
+}
+elseif (hasRequest('action') && getRequest('action') === 'test.stop') {
+	if (hasRequest('group_itemid')) {
+		$itemid = getRequest('group_itemid')[0];
+		$error = false;
+		try {
+			$sql_select = 'SELECT * FROM item_testing WHERE itemid='.$itemid;
```
Comment (sapajune) [inline frontends/php/items.php:919]: Should have SQL injection prevention.

File: frontends/php/include/schema.inc.php [hunk 3]
```diff
@@ -1704,6 +1704,270 @@
...
+				'length' => 1024,
+				'default' => '0',
+			],
+			'test_value' => [
+				'null' => false,
+				'type' => DB::FIELD_TYPE_TEXT,
+				'default' => '',
+			],
+		],
+	],
```
Comment (sapajune) [inline frontends/php/include/schema.inc.php:1975]: Creation statement for required additional table should be implemented somehow.

**Review by sapajune — APPROVED** [review]

**Review by sapajune — CHANGES_REQUESTED** [review]
See comments.

File: frontends/php/items.php (hunk 1 above)
Comment (PeterJokinen) [reply frontends/php/items.php]: This does indeed work for text values. Results in strings encased with single quotes.

Did some digging in Zabbix, I figured zabbix must have some sql injection prevention methods, and turns out it does. Found a function in zabbix([zbx_dbstr](https://github.com/zabbix/zabbix/blob/783cc48bba613df229b45fab9e6e486717d7e892/frontends/php/include/db.inc.php#L986)), that escapes strings for safe sql use. So I added checks for all string values for this sql.

Comment (sapajune) [reply frontends/php/items.php]: Good to know. That method handles escapes needed for different database drivers nicely.

File: frontends/php/items.php (hunk 2 above)
Comment (PeterJokinen) [reply frontends/php/items.php:919]: Handled by zabbix data type checks, as 'group_itemid' is an integer and cannot be changed to string. (check_fields function at line 267 ish)

File: frontends/php/include/schema.inc.php (hunk 3 above)
Comment (PeterJokinen) [reply frontends/php/include/schema.inc.php:1975]: Added table creation sql to the other [PR](https://github.com/digiaiiris/iiris/pull/1018), have a look there.

Comment (sapajune) [reply frontends/php/include/schema.inc.php:1975]: Works nicely for our Ansible setup. I suggest also including the table create script in this project (without USE row) with some comments/documentation.

Comment (PeterJokinen) [reply frontends/php/include/schema.inc.php:1975]: added table creation script to this PR also

**Comment by PeterJokinen** [comment]
Ping @bulkware @sapajune 
Changes has been made, ready for re-review

**Review by sapajune — APPROVED** [review]

**Review by bulkware — APPROVED** [review]

**Pull request merged**